contributing.md

```diff
@@ -0,0 +1,122 @@
+  end
+
+end
+```
+
+Let me explain a few things. The first command `config.vm.box = "handy-quickstart"` tells Vagrant that we want to use the base box that we just imported. The second command `config.vm.network :forwarded_port, guest: 3000, host: 3000` will make sure that when the Rails server serves a webpage on the VM, it will get passed on to the host. The commands in the `config.vm.provider` are performance tweaks. The first two lines solve connection issues Ubuntu VMs run into sometimes. It doesn't hurt to have it in there, and it will probably spare you some frustration. The last two lines let the VM grab a lot of resources from the host machine. This does *not* mean that the VM will hog 4 gigs of RAM and two cores all of the time. It just means that they *can* use those resources if push comes to shove.
+
+## Get Familiar with Your New VM
+
+ If you're not comfortable with Linux and the command line, you might want to take a break now to learn the basics. [Here](http://linuxcommand.org/lc3_lts0010.php) is a nice guide. You can get by if you're comfortable with the first three sections.
+
+ Now that you've created your Vagrantfile, you can `vagrant up` in your project directory and your VM will come to life! Get inside by `vagrant ssh`ing in from the project directory (wherever your Vagrantfile is). If you look at your `pwd`, you'll see that you were dropped in to `/home/vagrant` aka `/~`. This is not the same as the special `/vagrant` directory where we'll be spending most of our time. `cd` to `/vagrant`.
+
+ `/vagrant` is special because Vagrant _synchronizes_ this folder to the project directory on your host machine. So if you add a new file to your project directory, it will show up in `/vagrant` and vice versa. This will let you edit files on your host and run the server on the VM.
+
+## Create Config Files
+
+ It's time to grab Handy code. Clone the repo to `/vagrant` with `git clone https://github.com/Pitt-CSC/handy.git` (or the URL of your fork). If you jump the gun and try to start the server now, you'll see some donk-nasty error messages complaining about `twilio.yml`. handy needs `config/twilio.yml`, `config/secrets.yml`, and `database.yml` to work. But none of those files are there! That's because they contain senstive information that shouldn't be committed to GitHub. 
+
+ Helpfully, the project maintainers have provided sample versions of each file. To get started, you can just use `secrets.yml.sample` as is**. So make a new file called `secrets.yml` with the contents of `secrets.yml.sample`. For `database.yml.sample` we'll need to make one change. 
+
+ The first part looks like
+ ```
+default: &default
+  adapter: mysql2
+  encoding: utf8
+  pool: 5
+  username: root
+  password:
+ ```
+
+ Change it to
+ ```
+ default: &default
+  adapter: mysql2
+  encoding: utf8
+  pool: 5
+  username: root
+  password: root
+  ```
+ That's because when I configured MySQL on the base box I gave the root user the password `root`.
```

Comment:
Seems like a provisioning script could take care of all but the Twilio config. If not, we could provide a `config/database.yml.vagrant` file that can be copied without changes.